src/aws/mod.rs

```diff
@@ -124,6 +124,9 @@ pub trait ClientBuilder {
 
     /// Build the client using the given config settings.
     fn build(config: &SdkConfig) -> Self::Client;
+
+    /// Build the client using the given config settings and path style addressing.
+    fn build_and_force_path_style(config: &aws_types::SdkConfig) -> Self::Client;
```

Comment:
I'm not wild about having an S3 specific method on this generic trait.

What would think of, instead of adding this method here, adding a field to https://github.com/vectordotdev/vector/blob/475e93f182cc51e06ca74f8545a245203bcec1d0/src/common/s3.rs#L5 that controls the argument to `force_path_style`?

Reply:
Yeah, I wasn't crazy about this either, but lacking rust skills here didn't know how else to achieve this. Would that look something like this?

```
pub(crate) struct S3ClientBuilder;

// field here?

impl ClientBuilder for S3ClientBuilder {
```

Reply:
@jszwedko latest commit should address this! Lemme know what you think. Thanks